[Style] Migrate GridLength to using StyleLengthWrapper

weinig opened the pull request:
#### b4821ce36cfd1065d20f305cb1f1509e10f16c68
<pre>
[Style] Migrate GridLength to using StyleLengthWrapper
<a href="https://bugs.webkit.org/show_bug.cgi?id=295490">https://bugs.webkit.org/show_bug.cgi?id=295490</a>

Reviewed by Alan Baradlay.

Renames `GridLength` to `Style::GridTrackBreadth` to better match grammar
and replaces its use of `WebCore::Length` with a constrained style value
using `Style::LengthWrapperBase`.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/Length.h:
* Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp:
* Source/WebCore/rendering/GridTrackSizingAlgorithm.h:
* Source/WebCore/rendering/RenderGrid.cpp:
* Source/WebCore/rendering/style/GridLength.h: Removed.
* Source/WebCore/rendering/style/GridTrackSize.cpp:
* Source/WebCore/rendering/style/GridTrackSize.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleExtractorConverter.h:
* Source/WebCore/style/StyleExtractorSerializer.h:
* Source/WebCore/style/StyleInterpolationFunctions.h:
* Source/WebCore/style/StyleInterpolationWrappers.h:
* Source/WebCore/style/values/grid/StyleGridTrackBreadth.h: Added.

Canonical link: <a href="https://commits.webkit.org/297039@main">https://commits.webkit.org/297039@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9418223c40aa040fb836eb45cbb2ae6813377f14

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110379 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30038 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/20471 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116403 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/60633 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/112342 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30717 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38625 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/83931 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113327 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24510 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99391 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64372 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/23871 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/17531 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/60200 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/93891 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17587 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119195 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37419 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/27758 "Found 8 new test failures: accessibility/datetime/input-datetime-local-label-value.html http/tests/media/hls/hls-audio-tracks-language.html http/tests/media/hls/hls-webvtt-default.html http/tests/media/hls/hls-webvtt-style.html http/tests/media/hls/range-request-cross-origin.html http/tests/media/hls/track-in-band-multiple-cues.html http/tests/media/hls/track-webvtt-multitracks.html platform/mac/media/encrypted-media/fps-generateRequest.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/92900 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/37792 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95660 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92723 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23618 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37723 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15457 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33359 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37314 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/42785 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/36976 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40316 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/38684 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/rendering/style/GridTrackSize.h:
```diff
@@ -128,9 +142,9 @@ class GridTrackSize {
-    GridLength m_fitContentTrackBreadth;
+    Style::GridTrackBreadth m_minTrackBreadth;
+    Style::GridTrackBreadth m_maxTrackBreadth;
+    Style::GridTrackBreadth m_fitContentTrackBreadth;
```
Consider a default value?
```suggestion
    Style::GridTrackBreadth m_fitContentTrackBreadth { 0_css_px };
```

darinadler commented on Source/WebCore/style/values/grid/StyleGridTrackBreadth.h:
```diff
@@ -0,0 +1,150 @@
+    bool operator==(const GridTrackBreadth&) const = default;
+
+private:
+    GridTrackBreadthLength m_length;
+    Flex m_flex;
+
+    enum class GridTrackBreadthType : bool { Length, Flex };
+    GridTrackBreadthType m_type;
```
Consider defaults?
```suggestion
    GridTrackBreadthLength m_length { 0_css_px };
    Flex m_flex { 0_css_fr };

    enum class GridTrackBreadthType : bool { Length, Flex };
    GridTrackBreadthType m_type { GridTrackBreadthType::Length };
```

weinig committed:
[Style] Migrate GridLength to using StyleLengthWrapper
https://bugs.webkit.org/show_bug.cgi?id=295490

Reviewed by Alan Baradlay.

Renames `GridLength` to `Style::GridTrackBreadth` to better match grammar
and replaces its use of `WebCore::Length` with a constrained style value
using `Style::LengthWrapperBase`.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/Length.h:
* Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp:
* Source/WebCore/rendering/GridTrackSizingAlgorithm.h:
* Source/WebCore/rendering/RenderGrid.cpp:
* Source/WebCore/rendering/style/GridLength.h: Removed.
* Source/WebCore/rendering/style/GridTrackSize.cpp:
* Source/WebCore/rendering/style/GridTrackSize.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleExtractorConverter.h:
* Source/WebCore/style/StyleExtractorSerializer.h:
* Source/WebCore/style/StyleInterpolationFunctions.h:
* Source/WebCore/style/StyleInterpolationWrappers.h:
* Source/WebCore/style/values/grid/StyleGridTrackBreadth.h: Added.

Canonical link: https://commits.webkit.org/297039@main
